src/Sarif.Viewer.VisualStudio/Models/SarifErrorListItem.cs

```diff
@@ -59,7 +59,7 @@ public SarifErrorListItem(Run run, Result result, string logFilePath, ProjectNam
             }
 
             Tool = run.Tool.ToToolModel();
-            Rule = rule?.ToRuleModel(result.RuleId);
+            Rule = rule.ToRuleModel(result.RuleId);
```

Comment:
This is what fixes the bug. When `rule` is `null`, the `ToRuleModel` extension method synthesizes a `RuleModel` object whose `id` is the argument passed in (`result.RuleId`). But as the code stood, the extension method was never called when `rule` was `null`, and the `RuleModel` object ended up as `null`.

Reply:
Why is it ok for an extension method to behave this way on being passed a null item? That seems to break the principle of least surprise.



---
In reply to: [146093187](https://github.com/Microsoft/sarif-sdk/pull/646#discussion_r146093187) [](ancestors = 146093187)

Reply:
Should `ToRuleModel` run if rule is null?  It sounds like it's possible but very confusing considering how this differs from how methods of the class itself function.

---
In reply to: [146093187](https://github.com/Microsoft/sarif-sdk/pull/646#discussion_r146093187) [](ancestors = 146093187)

Reply:
@michaelcfanning I agree, `ToRuleModel` is oddly implemented. This is the smallest change that fixes the bug.